app/models/report.rb

```diff
@@ -14,4 +22,17 @@ def editable?(target_user)
   def created_on
     created_at.to_date
   end
+
+  private
+
+  def update_mentions
+    outgoing_mentions.destroy_all
+    mentioned_ids = extract_report_ids_from_content.map(&:to_i).uniq - [id]
+    available_report_ids = Report.where(id: mentioned_ids).pluck(:id)
+    mentioning_reports << Report.where(id: available_report_ids)
```

Comment:
これでよいのであれば、一旦idとして抽出しなくても `where` はさめばよさそうですね。
```suggestion
    mentioning_reports << Report.where(id: mentioned_ids)
```